js/mainApiMgr.js

```diff
@@ -47,6 +49,24 @@ function isValidWindow(event) {
     return result;
 }
 
+/**
+ * Method that is invoked when the application is reloaded/navigated
+ * window.addEventListener('beforeunload')
+ * @param windowName
+ */
+function reload(windowName) {
```

Comment:
Sorry I am bringing this up late, but we can use a name such as ```resetNecessities``` or ```sanitize``` or something along those lines. ```reload``` to me sounds like something else is happening.